sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -57,6 +61,27 @@ export function proxyPolicy(proxySettings?: ProxySettings): RequestPolicyFactory
   };
 }
 
+function extractAuthFromUrl(url: string) : { username?: string, password? : string, urlWithoutAuth: string } {
+  const atIndex = url.indexOf("@");
+  if (atIndex === -1) {
+    return { urlWithoutAuth: url};
+  }
+
+  const schemeIndex = url.indexOf("://")
```

Comment:
I wonder if you could get away with using a regex to simplify the parsing here:

```
const urlMatch = url.match(/((\w+):\/\/)?((.*):(.*)@)?.*$/);
const username = urlMatch[4];
const password = urlMatch[5];
```

Not sure if it's more likely to cause trouble than the current approach though, so take it or leave it :)

Reply:
I like it! It is more JavaScript-y.  Although I also need to remove the auth part from the original url. I'd need another regex to do that?

Reply:
Merging this.